packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -1,18 +1,281 @@
+import Textarea from 'primevue/textarea';
 import { WorkflowNode } from '@/types/workflow';
 import TeraDrilldown from '@/components/drilldown/tera-drilldown.vue';
-import { ModelConfigOperationState } from './model-config-operation';
+import TeraDrilldownSection from '@/components/drilldown/tera-drilldown-section.vue';
+import TeraDrilldownPreview from '@/components/drilldown/tera-drilldown-preview.vue';
+import TeraModelSemanticTables from '@/components/model/petrinet/tera-model-semantic-tables.vue';
+import { getModel } from '@/services/model';
+import {
+	createModelConfiguration,
+	getModelConfigurationById
+} from '@/services/model-configurations';
+import { Model, ModelConfiguration, Initial, ModelParameter } from '@/types/Types';
+import { ModelConfigOperation, ModelConfigOperationState } from './model-config-operation';
+import teraModelConfigEditor from './tera-model-config-editor.vue';
 
-defineProps<{
+const props = defineProps<{
 	node: WorkflowNode<ModelConfigOperationState>;
 }>();
-const emit = defineEmits(['close']);
+const emit = defineEmits([
+	'append-output-port',
+	'update-state',
+	'select-output',
+	'update-output-port',
+	'close'
+]);
+
+enum ConfigTabs {
+	Wizard = 'Wizard',
+	Notebook = 'Notebook'
+}
+
+const formSteps = ref([
+	{
+		label: 'Context'
+	},
+	{
+		label: 'Set values'
+	}
+]);
+
+const outputs = computed(() => {
+	if (!_.isEmpty(props.node.outputs)) {
+		return [
+			{
+				label: 'Select outputs to display in operator',
+				items: props.node.outputs
+			}
+		];
+	}
+	return [];
+});
+const selectedOutputId = ref<string>();
+const selectedConfigId = computed(
+	() => props.node.outputs?.find((o) => o.id === selectedOutputId.value)?.value?.[0]
+);
+
+const configCache = ref<Record<string, ModelConfiguration>>({});
+
+const activeIndex = ref<number>(0);
+const configName = ref<string>(props.node.state.configName);
+const configDescription = ref<string>(props.node.state.configDescription);
+const model = ref<Model>();
+
+const configInitials = ref<Initial[]>();
+const configParams = ref<ModelParameter[]>();
+
+onMounted(async () => {
+	const input = props.node.inputs[0];
+	if (input.value) {
+		const m = await getModel(input.value[0]);
+		if (m) {
+			model.value = m;
+
+			// TODO: set this only if configs don't exist already
+			// set the state with the model initials and params
+			const state = _.cloneDeep(props.node.state);
+			state.configInitials = m.semantics?.ode.initials;
+			state.configParams = m.semantics?.ode.parameters;
+			emit('update-state', state);
+		}
+	}
+});
+
+const updateState = (updatedField) => {
+	let state = _.cloneDeep(props.node.state);
+	state = { ...state, ...updatedField };
+	emit('update-state', state);
+};
+
+const updateConfigParam = (param) => {
+	const state = _.cloneDeep(props.node.state);
+	// find param with the same id and update it
+	state.configParams = state.configParams?.map((p) => {
+		if (p.id === param.id) {
+			return param;
+		}
+		return p;
+	});
```

Comment:
This looks a bit suspicious, are you updating from yourself - what is the intention here?

Reply:
Okay I see, the naming of things is a bit confusing, too many params. 

I think it would be clearer if we can do this:
```
const idx = state.configParams.findIndex(d => d.id === param.id);
if (idx) {
  state.configParams[idx] = param
}

```

Reply:
updated w ur suggestion